### ISSUE TITLE
Need to directly commit the wl_surface in some cases

### ISSUE THREAD

**Issue by Tamaranch** [wmww/gtk-layer-shell#143]
Since https://github.com/wmww/gtk-layer-shell/pull/52 the wl_surface is not committed directly but it is left to GTK to do it if necessary.

However, it seems that in some cases, like when negative (or large enough) margins are set so that the surface leaves the screen, a direct commit is needed to make it come back. See https://github.com/WayfireWM/wayfire/issues/1656.

Forcing a direct commit also solves a similar problem I reported for Labwc a little while ago: https://github.com/labwc/labwc/issues/499.

**Comment by wmww** [comment]
Yes, what's happening is the surface isn't getting a `.frame` wayland event, so GTK isn't committing. I think setting `_GdkWindowImplWayland::pending_commit` to TRUE (via the system we already have to access private GTK internals) might fix the problem, need to investigate more.

**Comment by RossComputerGuy** [comment]
I think I'm experiencing this issue. I am using Flutter with gtk-layer-shell to make a virtual keyboard and after the initial render, the surface isn't updating. Any updates on this?

**Comment by wmww** [comment]
@RossComputerGuy does the workaround in #167 help?

**Comment by RossComputerGuy** [comment]
I don't think I'm able to test that out since I am using the `bitsdojo_window` library which overrides certain things.
